### DISCUSSION TITLE
remove the array with duplicate element

### DISCUSSION THREAD

**Discussion by yiminghwang** [jax-ml/jax#12935] [Q&A] [answered]
Hi, there,

Is there any good way to remove the duplicated elements in DeviceArray? 
we all know it can be implemented as 
```python
a_list = [[1,0,0,1],[1,0,1,1],[0,1,1,0],[1,0,0,1],[1,0,1,1]]
res_list = [list(t) for t in set(tuple(list(_)) for _ in a_list)]
```
Is there any effecient implementation using JAX? directly use fori_loop to filter?

**Comment by jakevdp** [comment, marked as answer]
You can do this with [`jnp.unique`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.unique.html):
```python
import jax.numpy as jnp
a = jnp.array([[1,0,0,1],[1,0,1,1],[0,1,1,0],[1,0,0,1],[1,0,1,1]])
print(jnp.unique(a, axis=0))
# [[0 1 1 0]
#  [1 0 0 1]
#  [1 0 1 1]]
```

**Reply by yiminghwang** [reply]
Oh! thanks, jake!
